common/database/connector.php

```diff
@@ -0,0 +1,114 @@
+ *         throw $e;
+ *     }
+ * </code>
+ *
+ * Copyright (C) 2016 Matthew Vita <matthewvita48@gmail.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+namespace common\database;
+
+use \Doctrine\ORM\Tools\Setup;
+use \Doctrine\ORM\EntityManager;
+use \common\database\Auditor;
+
+final class Connector {
+    /**
+     * The pooled Doctrine connection.
+     */
+    public $entityManager;
+
+    /**
+     * Logger for noting connection/configuration information.
+     */
+    private $logger;
+
+    /**
+     * Default constructor.
+     */
+    private function __construct() {
+        $this->logger = new \common\logging\Logger("\common\database\Connector");
+        $this->createConnection();
+    }
+
+    /**
+     * The only public method for consumers to either create or gain access to the sole singleton
+     * instance of the class.
+     *
+     * @return Connector instance
+     */
+    public static function Instance() {
+        static $singletonInstance = null;
+        if ($singletonInstance === null) {
+            $singletonInstance = new Connector();
+        }
+
+        return $singletonInstance;
+    }
+
+    /**
+     * Creates the pooled Doctrine connection. All connection data is configurable via sqlconf.php.
+     * By default, the connection is pooled, in a nondev mode, and uses the pdo_mysql driver. Note
+     * that $GLOBALS["doctrine_connection_pooling"] and $GLOBALS["doctrine_dev_mode"] are used instead
+     * of $sqlconf[] because editing the sqlconf.php is not allowed (will mess up endusers trying to
+     * upgrade their install).
+     *
+     * @todo document throwables
+     */
+    private function createConnection() {
+        global $sqlconf;
+        $entityPath = array(__DIR__ . "../entities");
+
+        $this->logger->trace("Connecting with " . ($GLOBALS["doctrine_connection_pooling"] ? "pooled" : "non-pooled") . " mode");
+        $connection = array(
+            'driver'   => "pdo_mysql",
+            'host'     => $sqlconf["host"],
+            'port'     => $sqlconf["port"],
+            'user'     => $sqlconf["login"],
+            'password' => $sqlconf["pass"],
+            'dbname'   => $sqlconf["dbase"],
```

Comment:
Would it make sense to convert $sqlconf to a construction parameter to avoid needing to use a global?
Or is that overly complicated, since would then need to bring it in as a construction parameter for every services class?